File: lab8/result_file_manager.py

import struct

class ResultFileManager:
    def write_text(self, file_name: str, x: float, y: float):

        with open(file_name, 'w', encoding='utf-8') as f:
            f.write(f"x = {x}\n")
            f.write(f"y = {y}\n")

    def read_text(self, file_name: str):
        print("Змiст текстового файлу:")
        with open(file_name, 'r', encoding='utf-8') as f:
            for line in f:
                print(line.strip())

    def write_binary(self, file_name: str, x: float, y: float):
        with open(file_name, 'wb') as f:
            f.write(struct.pack('dd', x, y))

    def read_binary(self, file_name: str):
        print("Зчитано з двiйкового файлу:")
        with open(file_name, 'rb') as f:
            data = f.read(struct.calcsize('dd'))
            x, y = struct.unpack('dd', data)
            print(f"x = {x}")
            print(f"y = {y}")

File: lab5/ResultFileManager.java
package KI304FirchukLab5;

import java.io.*;

public class ResultFileManager {


    public void writeText(String fileName, double x, double y) throws IOException { //кинути виняток при помилці
        try (FileWriter writer = new FileWriter(fileName)) {
            writer.write("x = " + x + "\n");
            writer.write("y = " + y + "\n");
        }
    }

    public void readText(String fileName) throws IOException {
        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
            String line;
            System.out.println("Змiст текстового файлу:");
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }
        }
    }

    public void writeBinary(String fileName, double x, double y) throws IOException {
        try (DataOutputStream out = new DataOutputStream(new FileOutputStream(fileName))) {
            out.writeDouble(x);
            out.writeDouble(y);
        }
    }

    public void readBinary(String fileName) throws IOException {
        try (DataInputStream in = new DataInputStream(new FileInputStream(fileName))) {
            double x = in.readDouble();
            double y = in.readDouble();
            System.out.println("Зчитано з двiйкового файлу:");
            System.out.println("x = " + x);
            System.out.println("y = " + y);
        }
    }
}
